Lecture_2/Exercise_3/list.c

```diff
@@ -0,0 +1,147 @@
+#include "list.h"
+
+list* ListAdd(list **root_ptr, void *content_ptr)
+{
+        list *list_ptr = *root_ptr;
+
+	list *curr_item = (list*)malloc(sizeof(list));
+	curr_item->content = content_ptr;
+
+	if (list_ptr == NULL) {
+		curr_item->prev = NULL;
+		curr_item->next = NULL;
+
+		*root_ptr = curr_item;	// update root by first
+	} else {
+		list_ptr = ListGetLast(list_ptr);
+
+		curr_item->prev = list_ptr;
+		curr_item->next = list_ptr->next;
```

Comment:
Is there a way that on this line list_ptr->next will not be NULL? If no, then it makes sense to just write
`curr_item->next = NULL;`
for clarity.

Reply:
Pavlo, you are right) curr_item->next = NULL; is better/more clear in this case. Thanks!